src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -19,13 +19,47 @@
 from . import constants
 # from . import gls_fit           # used only if algorithm is "GLS"
 from . import ols_fit           # used only if algorithm is "OLS"
+from . import ramp_fit_class
 
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
 BUFSIZE = 1024 * 300000  # 300Mb cache size for data section
 
 
+def create_ramp_fit_class(model, dqflags=None):
+    """
+    Create an internal ramp fit class from a data model.
+
+    Parameters
+    ----------
+    model : data model
+        input data model, assumed to be of type RampModel
+
+    Return
+    ------
+    ramp_class : RampFitInternal
+        The internal ramp class.
+    """
+    ramp_class = ramp_fit_class.RampFitInternal()
+
+    ramp_class.set_arrays(
+        model.data, model.err, model.groupdq, model.pixeldq, model.int_times)
+
+    ramp_class.set_meta(
+        name=model.meta.instrument.name,
+        frame_time=model.meta.exposure.frame_time,
+        exp_ngroups=model.meta.exposure.ngroups,
+        group_time=model.meta.exposure.group_time,
+        groupgap=model.meta.exposure.groupgap,
+        nframes=model.meta.exposure.nframes,
+        drop_frames1=model.meta.exposure.drop_frames1)
+
+    ramp_class.set_dqflags(dqflags)
```

Comment:
Is there a reason to have this all called here instead of being bundled in `RampFitInternal.__init__()`?  I.e. can't this function be that init method?  Is there a reason to have it here?

Or should this be in the calling code from `RampFitStep`?  Will `jwst` and `romancal` construct this class in exactly the same way?  I suspect not, as `romancal` will have to broadcast their ndim=3 data to ndim=4.

Reply:
Thanks for catching the docstring.  I added it.